File: Asignaturas/Tercer Año/PDOO/Examenes/SolucionParcialPracticas2/P5-java/src/irrgarten/SuperPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author ismael
 */
public class SuperPlayer extends Player{
    public SuperPlayer(Player other){
        super(other);
    }
    
    @Override
    public float attack() {
        return (getStrength() + sumWeapons())*getIntelligence();
    }
    
    @Override
    public String toString() {
        String out = "[Soy Super]";
        return out + super.toString();
    }
    
    
}

File: Asignaturas/Tercer Año/PDOO/Examenes/SolucionParcialPracticas2/P5-Ruby/lib/irrgarten/super_player.rb
#encoding: UTF-8

require_relative 'player'
require_relative 'dice'

module Irrgarten
  class SuperPlayer < Player
  
    def initialize(other)
            copy(other)
    end
    
    def attack
            (Dice.intensity(@strength) + sumWeapons) * @intelligence
    end
    
    def to_s
            "[Soy Super] #{super.to_s} "
    end
    
  end
end
